CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1202,49 +1254,62 @@ function UIEditRoom:setDoorBlueprint(orig_x, orig_y, orig_wall)
     flags = 0
     y2 = y2 - 1
   end
-
-  self.blueprint_door.valid = checkDoorWalls(x, y, wall, self.room_type.swing_doors)
+  local world = self.ui.app.world
+  -- invalid_tile used to select the individual blueprint that is blocked
+  local invalid_tile = checkDoorWalls(x, y, wall, self.room_type.swing_doors)
+  self.blueprint_door.valid = invalid_tile == 0
   if self.blueprint_door.valid then
     -- Ensure that the door isn't being built on top of an object
-    local flag_names
-    if wall == "west" then
-      flag_names = {"buildableNorth", "buildableSouth"}
-    else
-      flag_names = {"buildableWest", "buildableEast"}
-    end
     local player_id = self.ui.hospital:getPlayerIndex()
-
-    if not validDoorTile(x, y, player_id, flag_names) or
-        not validDoorTile(x2, y2, player_id, flag_names) then
+    if not validDoorTile(x, y, player_id, orig_wall, world) or
+        not validDoorTile(x2, y2, player_id, orig_wall, world) then
       self.blueprint_door.valid = false
+      invalid_tile = 4
     end
     -- If we're making swing doors two more tiles need to be checked.
     if self.room_type.swing_doors then
       local dx = x_mod and 1 or 0
       local dy = y_mod and 1 or 0
-      if not validDoorTile(x + dx, y + dy, player_id, nil) or
-          not validDoorTile(x2 + dx, y2 + dy, player_id, nil) then
+      if not validDoorTile(x + dx, y + dy, player_id, orig_wall, world) or
+          not validDoorTile(x2 + dx, y2 + dy, player_id, orig_wall, world) then
         self.blueprint_door.valid = false
+        invalid_tile = invalid_tile | 8
       end
-      if not validDoorTile(x - dx, y - dy, player_id, nil) or
-          not validDoorTile(x2 - dx, y2 - dy, player_id, nil) then
+      if not validDoorTile(x - dx, y - dy, player_id, orig_wall, world) or
+          not validDoorTile(x2 - dx, y2 - dy, player_id, orig_wall, world) then
         self.blueprint_door.valid = false
+        invalid_tile = invalid_tile | 2
       end
     end
   end
-  if not self.blueprint_door.valid then
-    flags = flags + 16 -- Use red palette rather than normal palette
-  end
   if self.room_type.swing_doors then
-    for _, animation in ipairs(anim) do
-      animation:setAnimation(self.anims, 126, flags)
+    for i, animation in ipairs(anim) do
+    -- calculation here to flag blocked blueprint tiles
+      animation:setAnimation(self.anims, 126, flags + ((invalid_tile & 2^i)>0 and 1 or 0)*16)
```

Comment:
more bit tests.
Also, all operators have white space around them, "^", ">", and "*"